### PR TITLE
Limit NumPy version to pre-1.24

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
A lot of stuff breaks after update to numpy 1.24. This PR limits numpy vserion.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [X] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Review by stiepan — APPROVED** [review]

File: qa/setup_packages.py [hunk 1]
```diff
@@ -210,8 +210,9 @@ def get_install_string(self, idx, cuda_version=None):
             `cuda_version`: str, optional, default = None
                 Cuda version used for this query
         """
-        pkg_cmd = "{name}=={version}".format(name=self.get_name(cuda_version, idx),
-                                             version=self.get_version(idx, cuda_version))
+        version = version=self.get_version(idx, cuda_version)
```
Comment (JanuszL) [inline qa/setup_packages.py:213]: Can you update `print_configs` as well?

Comment (mzient) [reply qa/setup_packages.py:213]: Mhhh..... another CI respin :\

**Review by JanuszL — APPROVED** [review]

File: qa/setup_packages.py [hunk 2]
```diff
@@ -512,7 +514,8 @@ def print_configs(cuda_version):
         for v in pckg.get_all_versions(cuda_version):
             alias = BasePackage.get_alias(v)
             if alias is not None:
-                v = "{}=={}".format(alias, v)
+                op = "" if str(v)[0] in ("<", ">", "=") else "=="
```
Comment (stiepan) [inline qa/setup_packages.py:517]: nitpick, you can go with 
```suggestion
                op = "" if str(v)[0] in "<>=") else "=="
```

Comment (mzient) [reply qa/setup_packages.py:517]: I know, it was a deliberate choice.

**Review by JanuszL — APPROVED** [review]

**Review by stiepan — APPROVED** [review]

**Review by JanuszL — APPROVED** [review]

**Pull request merged**